internal/error.go

```diff
@@ -489,6 +495,7 @@ func (wc *workflowEnvironmentInterceptor) NewContinueAsNewError(
 		WorkflowRunTimeout:       options.WorkflowRunTimeout,
 		WorkflowTaskTimeout:      options.WorkflowTaskTimeout,
 		VersioningIntent:         options.VersioningIntent,
+		RetryPolicy:              options.RetryPolicy,
```

Comment:
This appears to be a breaking change since `options.RetryPolicy` can also be set by `workflow.WithChildWorkflowOptions`. Workflows may be setting a retry policy for a child workflow and now we may also override the parents. This a problem with other options as well, but now we are spreading it to `RetryPolicy` as well. I will need to discuss with the team if this is an acceptable change.

Reply:
Yes, I fully understand the justification. We just take backwards compatibility very seriously and don't want to break any previously working workflows.

Reply:
Your change will now cause that `RetryPolicy` in `WithChildWorkflowOptions ` to be used as the new `RetryPolicy` on continue as new I believe.

Reply:
I think the issue is if you had these steps:

* Set `WithChildWorkflowOptions`
* Execute child workflow
* Return continue-as-new error with no retry policy

The behavior before your change was to have continue-as-new not set a retry policy. But now after this change, the retry policy will come from `WithChildWorkflowOptions` which is an incompatible change from what it did before which was no retry policy set from context options.

Reply:
> I think the scenarios you mentioned are covered by

The scenario I mention combines setting retry policy for child workflow and also returning continue as new.

Take this workflow (typed in GH so I may have typos):

```go
func MyWorkflow(ctx context.Context) error {
	ctx = workflow.WithChildOptions(workflow.ChildWorkflowOptions{
		RetryPolicy: &temporal.RetryPolicy{MaximumAttempts: 5},
	})
	err := workflow.ExecuteChildWorkflow(ctx, MyChildWorkflow).Get(ctx, nil)
	if err != nil {
		return err
	}
	// This never used to use the retry policy before, but now it does
	return workflow.NewContinueAsNewError(ctx, MyWorkflow)
}
```

Before your change (i.e. how it works today), you would not be setting a retry policy on continue as new (which means it won't retry for workflows by default). But after your change, now continue as new has a retry policy of max-attempts 5. Unless I'm misunderstanding somewhere, this is the backwards incompatible change being referenced.

Reply:
> Fortunately, this issue can be easily worked around by using a different context, as illustrated in TestContinueAsNewWithWithChildWF.

We have a standing rule not to break behavior compatibility of existing code on things like this. We can't just ask people to change code that works for them today and silently start using retry policy set for something else on continue as new.

We could make a `NewContinueAsNewErrorWithOptions` that accepts some options, but they might as well instantiate `&workflow.ContinueAsNewError{...}` themselves.